sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -720,8 +721,14 @@ case class DescribeColumnCommand(
     }
 
     val catalogTable = catalog.getTempViewOrPermanentTableMetadata(table)
-    val colStats = catalogTable.stats.map(_.colStats).getOrElse(Map.empty)
-    val cs = colStats.get(field.name)
+    val colStats = if (conf.caseSensitiveAnalysis) {
+      catalogTable.stats.map(_.colStats).getOrElse(Map.empty)
+    } else {
+      catalogTable.stats.map(_.colStats.map {
+        case (key, value) => key.toLowerCase(Locale.ROOT) -> value
+      }).getOrElse(Map.empty)
+    }
+    val cs = colStats.get(getColumnName(field.name))
```

Comment:
Can we just do:

```scala
val colStatsMap = catalogTable.stats.map(_.colStats).getOrElse(Map.empty)
val colStats = if (conf.caseSensitiveAnalysis) colStatsMap else CaseInsensitiveMap(colStatsMap)
```

? I think this will fix all problems here.

Reply:
Ur, I totally forgot `CaseInsensitiveMap`... nice suggestion.